lib/Runtime/Types/PathTypeHandler.cpp

```diff
@@ -2156,43 +2155,54 @@ namespace Js
         return nextType;
     }
 
-    ObjectSlotAttributes * PathTypeHandlerBase::UpdateAttributes(Recycler * recycler, ObjectSlotAttributes * oldAttributes, uint8 oldPathSize, uint8 newTypePathSize)
+    ObjectSlotAttributes * PathTypeHandlerBase::UpdateAttributes(Recycler * recycler, ObjectSlotAttributes * oldAttributes, uint8 newTypePathSize)
     {
         ObjectSlotAttributes * newAttributes = RecyclerNewArrayLeaf(recycler, ObjectSlotAttributes, newTypePathSize);
-        uint8 initStart;
+        uint16 initStart;
         if (oldAttributes == nullptr)
         {
             initStart = 0;
         }
         else
         {
-            // In branching cases, the new type path may be shorter than the old.
-            initStart = min(newTypePathSize, oldPathSize);
+            // Copy the portion of the old attributes that refers to the head of the path
+            initStart = GetPathLength();
+            Assert(newTypePathSize >= initStart);
+            __analysis_assume(newTypePathSize >= initStart);
```

Comment:
nit: you can use this macro AnalysisAssert, which does both.

Reply:
Good suggestion, thanks.